.github/workflows/code-quality-tests.yaml

```
@@ -0,0 +1,53 @@
+name: Code Quality Tests
+
+on:
+  push:
+    branches:
+      - '**'
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
```

Using `paths` or `paths-ignore` with `push` does not work as you would expect it to, use `pull_request` so that the whole diff is made accordingly:

```suggestion
  pull_request:
    branches:
      - main
    paths-ignore:
      - 'docs/**'
      - '*.md'
```

---

> Using paths or paths-ignore with push does not work as you would expect it to
How do you think it works?

---

it will run for every push as long as a you have a PR open. (and you can always open a draft pr early to get some builds if needed).

---

Thanks for the headsup and "Request Changes", lesson learned. :)